PR: Improve style and UX of new toolbar in collections viewer (Variable Explorer)

ccordoba12 opened the pull request:
## Description of Changes

- This is a follow up of PR #17473 to improve the style of the new toolbar by using the same stylesheet of pane toolbars and reorganize its icons a bit.
- Add new icons for `Insert above`/`Insert below` actions (they were using the same as the `Insert` action).
- Set disabled state color for SVG icons. Before buttons associated to those icons appeared active even if they were disabled.
- Enable/disable buttons in the toolbar according to certain conditions (e.g. is there something selected and in the same row? Is it read only?)
- Improve the context menu of collections and the Variable Explorer.

----

**Before**

![image](https://user-images.githubusercontent.com/365293/166482556-f76b5c21-18bd-4435-88fa-1081976a0f7e.png)

![image](https://user-images.githubusercontent.com/365293/166482793-938b0f52-4524-47b2-8fdd-02303e6b0491.png)

![imagen](https://user-images.githubusercontent.com/365293/168480981-1e9e7ead-7211-4072-8322-bab39f5620f5.png)

----

**After**

![image](https://user-images.githubusercontent.com/365293/166481705-3b91b03f-690b-42f5-a6b5-c536e114dfc7.png)

![image](https://user-images.githubusercontent.com/365293/166482095-8490a05f-8302-4ed8-b519-a62d551f711e.png)

![imagen](https://user-images.githubusercontent.com/365293/168480935-2688e242-6798-47d2-94da-34ae3e255877.png)

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

<!--- TYPE YOUR USER/NAME AFTER THE FOLLOWING: --->
I certify the above statement is true and correct:

<!--- Thanks for your help making Spyder better for everyone! --->

ccordoba12 commented:
@CAM-Gerlach, this is the way I decided to order the context menu/toolbar actions of that viewer:

```python
self.menu_actions = [
    self.edit_action,
    self.copy_action,
    self.paste_action,
    self.rename_action,
    self.remove_action,
    self.save_array_action,
    MENU_SEPARATOR,
    self.insert_action,
    self.insert_action_above,
    self.insert_action_below,
    self.duplicate_action,
    MENU_SEPARATOR,
    self.view_action,
    self.plot_action,
    self.hist_action,
    self.imshow_action,
    MENU_SEPARATOR,
    resize_action,
    resize_columns_action
]
```

Let me know if you have suggestions/improvements to it.

dalthviz reviewed (commented):
Thanks @ccordoba12 ! Checked locally and seems like the insert above/below options are not working:

![toolbar](https://user-images.githubusercontent.com/16781833/167681910-0930b8f9-9b1b-40e8-939c-044f7d949632.gif)

Other than that this LGTM :+1:

CAM-Gerlach commented:
> @CAM-Gerlach, this is the way I decided to order the context menu/toolbar actions of that viewer:

Sorry for the delay, I was preparing for and helping attend/organize at PyCon and then trying to catch up on everything after. That looks really good to me :+1:

ccordoba12 commented:
@dalthviz and @CAM-Gerlach, thanks for the review!

@dalthviz, I fixed the error you reported above by reverting some changes I did in a previous commit. This should be ready now.

dalthviz commented:
Checked again and although now the insert buttons work, seems like an update to the toolbuttons/actions state is missing when duplicating or removing elements from the collection (after such actions no element is visibly select yet all the toolbuttons stay activated without working):


![toolbar2](https://user-images.githubusercontent.com/16781833/168110867-d4d62153-2387-4047-8926-26ea7e5466de.gif)

To go again to a coherent state you need to select something

dalthviz commented:
Just in case, this needs now a rebase to get the fixes done at #17860

dalthviz reviewed (approved):
Thanks @ccordoba12 !

dalthviz commented:
Seems like the fast test are failing at `spyder/widgets/tests/test_collectioneditor.py::test_remove_remote_variable` with a segfault 🤔

ccordoba12 commented:
@dalthviz, I fixed the segfault by simply avoiding to deselect the current selection when running our tests.

Besides, I made the context menu of the Variable Explorer to match the new one for collections for consistency. I added a new screenshot about that to the description.

dalthviz reviewed (approved):
Thanks @ccordoba12 !
